### PR TITLE
Replace Poetry with uv for dependency management

### LINKED ISSUE #1577: [FEATURE] Add py-spy to dependencies
## Is your feature request related to a problem? Please describe.

- py-spy now supports Python 3.12.
  - https://github.com/benfred/py-spy/releases/tag/v0.4.0

## Describe the solution you'd like

- Add py-spy to dependencies

### LINKED ISSUE #1578: [FEATURE] Replace poetry with uv
## Is your feature request related to a problem? Please describe.

- We would like to use uv as the package manager.
  - https://docs.astral.sh/uv/ 

## Describe the solution you'd like

- Replace poetry with uv

### PR DESCRIPTION
**Pull request by purplesmoke05**
- Close #1578 
- Close #1577 
- Upgrade pydantic

### CODE REVIEW

File: Dockerfile
Comment (YoshihitoAso) [inline Dockerfile:1]: uv&python のインストールがなさそうです

Comment (purplesmoke05) [reply Dockerfile:1]: Thank you. I fixed it in this commit: https://github.com/BoostryJP/ibet-Wallet-API/pull/1579/commits/3a4afc98d356105af4e5d64ae9e539b6e98bafc3

```bash
$ docker run -it ibet-wallet-api:uv bash --login -c "python -V"
Python 3.12.2
```

Comment (YoshihitoAso) [reply Dockerfile:1]: One more thing. You will probably need --all-extras or similar when using uv sync.

Comment (purplesmoke05) [reply Dockerfile:1]: Appreceiate your consistent feedback. I fixed that in this commit: https://github.com/BoostryJP/ibet-Wallet-API/pull/1579/commits/362d075a4b44ee2df28e024ab7b18efdcd603941

I have also confirmed that the extra package works as expected.

```bash
docker run -it ibet-wallet-api:uv bash --login -c "ibet-explorer --help"
 Usage: ibet-explorer [OPTIONS]

╭─ Options ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ --url                       TEXT     ibet-Wallet-API server URL to connect [default: http://localhost:5000]                                                                                                                                                                │
│ --lot-size                  INTEGER  Lot size to fetch Block Data list [default: 100]                                                                                                                                                                                      │
│ --install-completion                 Install completion for the current shell.                                                                                                                                                                                             │
│ --show-completion                    Show completion for the current shell, to copy it or customize the installation.                                                                                                                                                      │
│ --help                               Show this message and exit.                                                                                                                                                                                                           │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

**Comment by purplesmoke05** [comment]
This PR is now ready for review ✅

**Review by YoshihitoAso — APPROVED** [review]

**Pull request merged**